packages/utils/src/format.ts

```diff
@@ -27,3 +28,24 @@ export function truncBytes(root: Uint8Array | string): string {
   const str = typeof root === "string" ? root : toHexString(root);
   return str.slice(0, 14);
 }
+
+/**
+ * Formats an address according to [ERC55](https://eips.ethereum.org/EIPS/eip-55)
+ *
+ * @param address an hex address
+ * @returns an ERC55 formatted version of `address`
+ */
+export function toChecksumAddress(address: string): string {
+  const rawAddress = address.toLowerCase().startsWith("0x") ? address.slice(2) : address;
```

Comment:
could add some sanity check here and throw if it not a valid address, might wanna at least make sure the length is correct

Reply:
Do you know of existing utility providing that in the codebase?

Reply:
Not sure, can just check what web3-utils does in terms of pre-validation